File: striver/S1L5/python/rev.py
#
#   method 1 = new array

# size = int(input())
# arr = [0]*size
# for i in range(0,size):
#     arr[i]=int(input())
# revarr = [0]*size
# for i in range(0,size):
#     revarr[i]=arr[size-1-i]
# for i in range(0,size):
#     print(revarr[i],end=" ")

# method 2 = by two pointer approach

# size = int(input())
# arr = [0]*size
# for i in range(0,size):
#     arr[i]=int(input())
# l = int(0)
# r = int(size-1)
# while l<r :
#     temp = arr[r]
#     arr[r] = arr[l]
#     arr[l] = temp
#     l=l+1
#     r=r-1
# for i in range(0,size):
#     print(arr[i],end=" ")

# method 3 = by recusive approach

size = int(input())
arr = [0]*size
for i in range(0,size):
    arr[i]=int(input())
def revarr(arr,l,r):
    if l>=r :
        return
    temp = arr[r]
    arr[r] = arr[l]
    arr[l] = temp
    
    return revarr(arr,l=l+1,r=r-1)
revarr(arr,0,size-1)
for i in range(0,size):
    print(arr[i],end=" ")

File: striver/S1L5/java/rev.java
package S1L5.java;

import java.util.Scanner;

public class rev {
    public static void main(String[] args) {
        //       method1 == by using new array

        //Scanner sc = new Scanner(System.in);
        // int size = sc.nextInt();
        // int[] arr = new int[size];
        // for(int i=0;i<size;i++){
        //     arr[i] = sc.nextInt();
        // }
        // int[] revarr = new int[size];
        // for(int i=0;i<size;i++){
        //     revarr[i]=arr[size-i-1];
        // }
        // for(int i=0;i<size;i++){
        //     System.out.print(revarr[i]+" ");
        // }
        //

        //        method2 == two pointer approach
        
        // Scanner sc = new Scanner(System.in);
        // int size = sc.nextInt();
        // int[] arr = new int[size];
        // for(int i=0;i<size;i++){
        //     arr[i] = sc.nextInt();
        // }
        // int l=0,r=size-1;
        // while(l<r){
        //     int temp = arr[r];
        //     arr[r] = arr[l];
        //     arr[l] = temp;
        //     l++;
        //     r--;
        // }
        // for(int i=0;i<size;i++){
        //     System.out.print(arr[i]+" ");
        // }

        //  method3 = recursiveapproach

        Scanner sc = new Scanner(System.in);
        int size = sc.nextInt();
        int[] arr = new int[size];
        for(int i=0;i<size;i++){
            arr[i] = sc.nextInt();
        }
        revarr(arr,0,size-1);
        for(int i=0;i<size;i++){
            System.out.print(arr[i]+" ");
        }


    }
    public static void revarr(int[] arr,int l,int r){
        if(l>=r) return;

        int temp = arr[r];
        arr[r] = arr[l];
        arr[l] = temp;
        
        revarr(arr, ++l, --r);
    }
}
